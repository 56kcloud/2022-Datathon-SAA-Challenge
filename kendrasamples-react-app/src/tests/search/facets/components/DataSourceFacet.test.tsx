import { shallow } from "enzyme";
import React from "react";
import { DataSourceFacet } from "../../../../search/facets/components/DataSourceFacet";
import {
  mockDataSourceIdValueCountPairs,
  mockDataSourceNameLookup,
} from "../../../mock/mock";
import _ from "lodash";

const facetConfiguration = {
  facetsToShowWhenUncollapsed: 5,
  showCount: true,
};

describe("A <DataSourceFacet />", () => {
  it("renders correctly when data source lookup not provided", () => {
    const component = shallow(
      <DataSourceFacet
        valueCountPairs={mockDataSourceIdValueCountPairs}
        dataSourceNameLookup={undefined}
        disableAdd={false}
        facetConfiguration={facetConfiguration}
        onSelectionChange={_.noop}
        onClear={_.noop}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("renders correctly when data source lookup provided", () => {
    const component = shallow(
      <DataSourceFacet
        valueCountPairs={mockDataSourceIdValueCountPairs}
        dataSourceNameLookup={mockDataSourceNameLookup}
        disableAdd={false}
        facetConfiguration={facetConfiguration}
        onSelectionChange={_.noop}
        onClear={_.noop}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
