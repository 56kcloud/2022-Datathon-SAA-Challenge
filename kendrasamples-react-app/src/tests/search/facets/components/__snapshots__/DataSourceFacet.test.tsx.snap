// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A <DataSourceFacet /> renders correctly when data source lookup not provided 1`] = `""`;

exports[`A <DataSourceFacet /> renders correctly when data source lookup provided 1`] = `
<Facet
  attributeName="_data_source_id"
  disableAdd={false}
  facetConfiguration={
    Object {
      "facetsToShowWhenUncollapsed": 5,
      "showCount": true,
    }
  }
  onClear={[Function]}
  onSelectionChange={[Function]}
  values={
    Array [
      Object {
        "Count": 10,
        "DocumentAttributeValue": Object {
          "StringValue": "mock-data-source-id",
        },
        "ValueLabel": "MockDataSourceName",
      },
    ]
  }
/>
`;
